test/jdk/java/awt/TrayIcon/ActionCommand/ActionCommand.java

```diff
@@ -44,15 +58,19 @@ public class ActionCommand {
     TrayIcon icon;
     ExtendedRobot robot;
 
-    boolean actionPerformed = false;
-    Object actionLock = new Object();
-    String actionCommand = null;
+    volatile boolean actionPerformed = false;
+    volatile String actionCommand = null;
+    final Object actionLock = new Object();
+
     static boolean isMacOS = false;
 
     public static void main(String[] args) throws Exception {
-        if (! SystemTray.isSupported()) {
-            System.out.println("SystemTray not supported on the platform under test. " +
-                    "Marking the test passed");
+        if (Platform.isOnWayland()) {
+            // The current robot implementation does not support
+            // clicking in the system tray area.
+            throw new SkippedException("Skipped on Wayland");
+        } else if (!SystemTray.isSupported()) {
+            throw new SkippedException("SystemTray is not supported on this platform.");
         } else {
             if (System.getProperty("os.name").toLowerCase().startsWith("win")) {
```

Comment:
`Platform.isWindows()` could work better… as well `Platform.isOSX()` for Mac.

The case `SystemTrayIconHelper.isOel7orLater()` should probably also throw `SkippedException`.